backend/module/chaos/serverexperimentation/xds/xds_cache.go

```diff
@@ -64,10 +64,21 @@ func refreshCache(ctx context.Context, storer experimentstore.Storer, snapshotCa
 		if _, exist := clusterFaultMap[cluster]; !exist {
 			logger.Debugw("Removing experiments for cluster", "cluster", cluster)
 
-			// in order to remove fault, we need to set the snapshot with empty runtime resource
-			emptyRuntimeResource := generateRTDSResource([]*experimentation.Experiment{}, rtdsConfig, ttl, logger)
+			var emptyResource []gcpTypes.ResourceWithTtl
+			ecdsClusterEnabled := containsString(ecdsConfig.enabledClusters, cluster) || containsString(ecdsConfig.enabledClusters, "*")
+			if ecdsClusterEnabled {
+				// in order to remove fault, we need to set the snapshot with default ecds config
+				emptyResource = generateEmptyECDSResource(cluster, ecdsConfig, logger)
```

Comment:
Wouldn't we set this for all clusters? Otherwise clusters that have the ECDS filter configured but that have not been enabled in Clutch will be stuck waiting for the init timeout